zeebe/exporters/elasticsearch-exporter/src/test/java/io/camunda/zeebe/exporter/SchemaManagerIT.java

```diff
@@ -0,0 +1,106 @@
+import org.junit.jupiter.api.Test;
+import org.mockito.MockedStatic;
+import org.testcontainers.elasticsearch.ElasticsearchContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+@Testcontainers
+public class SchemaManagerIT {
+  @Container
+  private static final ElasticsearchContainer CONTAINER =
+      TestSupport.createDefaultContainer().withEnv("action.destructive_requires_name", "false");
+
+  private static final ElasticsearchExporterConfiguration CONFIG =
+      new ElasticsearchExporterConfiguration();
+
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  @BeforeAll
+  static void setup() {
+    CONFIG.url = "http://" + CONTAINER.getHttpHostAddress();
+  }
+
+  @Test
+  public void shouldUseVersionedComponentTemplateForIndexTemplates() throws IOException {
+    // given
+    final var restClient = RestClientFactory.of(CONFIG);
+    final var esClient = new ElasticsearchClient(CONFIG, new SimpleMeterRegistry());
+
+    // when
+    // we create an 8.5 component template
+    final var schemaManager = new SchemaManager(esClient, CONFIG);
+    schemaManager.createSchema("8.5.0");
+
+    // create the 8.6 component template
+    final var upgradedVersion = "8.6.0";
+    try (final MockedStatic<VersionUtil> mockedVersionUtil = mockStatic(VersionUtil.class)) {
+      mockedVersionUtil.when(VersionUtil::getVersion).thenReturn(upgradedVersion);
+      mockedVersionUtil.when(VersionUtil::getVersionLowerCase).thenReturn(upgradedVersion);
+
+      final var schemaManager2 = new SchemaManager(esClient, CONFIG);
+      schemaManager2.createSchema("8.6.0");
+    }
+
+    // Use a new schema manager to mock 8.5 broker restart which then triggers schema manager.
+    // 8.5 schema manager runs again and attempts to create old component template
+    final var schemaManager3 = new SchemaManager(esClient, CONFIG);
+    schemaManager3.createSchema("8.5.0");
+
+    // then
+    // 8.6.0 component template exists
+    final var request = new Request("GET", "/_component_template/" + CONFIG.index.prefix + "*");
+    final var response = restClient.performRequest(request);
+    final var componentTemplateInElasticsearch =
+        MAPPER.readValue(response.getEntity().getContent().readAllBytes(), JsonNode.class);
+
+    assertThat(componentTemplateInElasticsearch.at("/component_templates"))
+        .anySatisfy(
+            node ->
+                assertThat(node.at("/name").asText()).isEqualTo("zeebe-record-" + upgradedVersion));
+
+    // all 8.6.0 index templates are composed of the new index template
+    final var templateReq = new Request("GET", "/_index_template/*" + upgradedVersion + "*");
+    final var templateRes = restClient.performRequest(templateReq);
+    final var indexTemplates =
+        MAPPER.readValue(templateRes.getEntity().getContent().readAllBytes(), JsonNode.class);
+
+    assertThat(indexTemplates.at("/index_templates"))
```

Comment:
can we add a `isNotEmpty()` to this assertion chain? I believe that `allSatisfy` will succeed on an empty list